apps/mymedicare_cb/views.py

```diff
@@ -0,0 +1,149 @@
+    redirect_uri = getattr(settings, 'SLS_REDIRECT_URI',
+                           'http://localhost:8000/mymedicare/sls-callback')
+    userinfo_endpoint = getattr(
+        settings, 'SLS_USERINFO_ENDPOINT', 'https://dev.accounts.cms.gov/v1/oauth/userinfo')
+    verify_ssl = getattr(settings, 'SLS_VERIFY_SSL', False)
+    code = request.GET.get('code')
+    state = request.GET.get('state')
+    aus = AnonUserState.objects.get(state=state)
+    next_uri = aus.next_uri
+    token_dict = {
+        "grant_type": "authorization_code",
+        "code": code,
+        "redirect_uri": redirect_uri}
+
+    # Call SLS token api: ", "https://dev.accounts.cms.gov/v1/oauth/token"  as
+    # a POST
+    r = requests.post(token_endpoint, json=token_dict, verify=verify_ssl)
+    token_response = {}
+    if r.status_code != 200:
+        return HttpResponse("An unknown error has occurred.", status=500)
+
+    token_response = r.json()
+    # Create the Bearer
+    bt = "Bearer %s" % (token_response['access_token'])
+    # build a headers dict with Authorization
+    headers = {"Authorization": bt}
+    # Call SLS userinfo: ",
+    # "https://dev.api.bluebutton.cms.gov/v1/oauth/userinfo  as a GET with
+    # Authorization Bearer token in header.
+    r = requests.get(userinfo_endpoint, headers=headers, verify=verify_ssl)
+    # print("Status", r.status_code)
+    if r.status_code != 200:
+        return HttpResponse("An unknown error has occurred.", status=500)
+    # Get the userinfo response object
+    user_info = r.json()
+    try:
+        user = User.objects.get(username=user_info['sub'])
+        if not user.first_name:
+            user.first_name = user_info['given_name']
+        if not user.last_name:
+            user.last_name = user_info['family_name']
+        if not user.email:
+            user.email = user_info['email']
+        user.save()
+    except User.DoesNotExist:
+        # Create a new user. Note that we can set password
+        # to anything, because it won't be checked.
+        user = User(username=user_info['sub'], password='',
+                    first_name=user_info['given_name'],
+                    last_name=user_info['family_name'],
+                    email=user_info['email'])
+        user.save()
+    up, created = UserProfile.objects.get_or_create(
+        user=user, user_type='BEN')
+    group = Group.objects.get(name='BlueButton')
+    user.groups.add(group)
+    # Log in the user
+    user.backend = 'django.contrib.auth.backends.ModelBackend'
+    login(request, user)
+    # Determine patient_id
+    fhir_source = get_resourcerouter()
+    cx, g_o_c = Crosswalk.objects.get_or_create(
+        user=user, fhir_source=fhir_source)
+    hicn = user_info.get('hicn', "999999999A")
+    cx.user_id_hash = hicn
+    cx.save()
+    # URL for patient ID.
+    url = fhir_source.fhir_url + \
+        "Patient/?identifier=http%3A%2F%2Fbluebutton.cms.hhs.gov%2Fidentifier%23hicnHash%7C" + \
```

Comment:
Should we set the identifier string as a settings value?
We can then set a value for each environment.